### PR TITLE
Get method descriptor from ptypes and rtype

### PR DESCRIPTION
**Pull request by liqunl**
With MethodType replaced with OpenJDK version, method descriptor string
will be lazy initialized. This commit will get the method descriptor
from the ptypes and rtype.

Signed-off-by: Liqun Liu <liqunl@ca.ibm.com>

### CODE REVIEW

**Comment by liqunl** [comment]
Depends on #7628 and #7625

**Comment by liqunl** [comment]
@andrewcraik May I have your review?

**Comment by DanHeidinga** [comment]
Part of #7352

**Comment by liqunl** [comment]
@cathyzhyi Since Andrew's not around, could you review this PR?

File: runtime/compiler/env/j9method.cpp [hunk 1]
```diff
@@ -7454,6 +7454,91 @@ static TR::DataType typeFromSig(char sig)
    return TR::NoType;
    }
 
+char* sigForPrimitiveOrVoid(TR_OpaqueClassBlock* clazz)
```
Comment (cathyzhyi) [inline runtime/compiler/env/j9method.cpp]: can you add `static` keyword so that it's clearer the method is only accessible in this file?

**Comment by cathyzhyi** [comment]
LGTM other than one small suggestion.

**Comment by liqunl** [comment]
@fengxue-IS Could you rebase #7628 and #7625 so that I can test the changes again?

**Comment by genie-openj9** [comment]
Can one of the admins verify this patch?

**Comment by 0xdaryl** [comment]
No longer relevant.

**Pull request closed without merging**